crates/ruff_linter/src/rules/ruff/rules/missing_fstring_syntax.rs

```diff
@@ -94,20 +94,32 @@ fn should_be_fstring(
     };
 
     let mut kwargs = vec![];
+    let mut last_expr: Option<&ast::Expr> = None;
     for expr in semantic.current_expressions() {
         match expr {
             ast::Expr::Call(ast::ExprCall {
                 arguments: ast::Arguments { keywords, .. },
                 func,
                 ..
             }) => {
-                if let ast::Expr::Attribute(ast::ExprAttribute { .. }) = func.as_ref() {
-                    return false;
+                if let ast::Expr::Attribute(ast::ExprAttribute { value, .. }) = func.as_ref() {
+                    match value.as_ref() {
+                        ast::Expr::StringLiteral(expr_literal)
+                            if expr_literal.value.as_slice().contains(literal) =>
+                        {
+                            return false;
+                        }
+                        value if last_expr == Some(value) => {
+                            return false;
+                        }
+                        _ => {}
```

Comment:
What if we have `"{test}".attribute.chaining.call()"` Should this be excluded too or is this handled by the `value if last_expr == Some(value)`? If so, it might be worth adding some comments explaining the cases these branches handle

Reply:
> Should this be excluded too or is this handled by the value if `last_expr == Some(value)`?

I just tested this, and it is handled by that case! I'll add some clarification though, that's a good idea.